app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2162,6 +2167,15 @@ class BrowserTabFragment :
         webShareRequest.launch(JsCallbackData(data, featureName, method, id))
     }
 
+    private fun screenLock(featureName: String, method: String, id: String, data: JSONObject) {
+        val returnData = JsOrientationHandler().updateOrientation(JsCallbackData(data, featureName, method, id), activity)
```

Comment:
You want to inject the JsOrientationHandler rather than creating a new instance every time not only because of that but for testing purposes.

Reply:
For sure, this is one of those ugly-draft things. Will tackle it together with unit tests. I'd like to settle on what `JsOrientationHandler` looks like first 👍